### PR TITLE
Origami Upgrade

### PR DESCRIPTION
**Pull request by edds**
Upgrade origami components and `n-ui-foundations` as part of the breaking change. This is mostly just swapping mixins to the new names.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "n-ui-foundations": "^3.0.0",
-    "o-quote": "^2.0.0"
+    "n-ui-foundations": "^v4.0.0-beta.1",
```
Comment (chee) [inline bower.json]:
```suggestion
    "n-ui-foundations": "^4.0.0-beta.1",
```

File: bower.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
...
+    "o-quote": "^4.0.0",
+    "o-editorial-typography": "^v1.0.0"
```
Comment (chee) [inline bower.json]:
```suggestion
    "o-editorial-typography": "^1.0.0"
```

File: scss/_blockquotes.scss [hunk 3]
```diff
@@ -1,14 +1,6 @@
...
-	@include oQuoteStandard();
-	@include oTypographyMargin($bottom: 7);
-
-	cite {
-		@include oQuoteCite();
-		@include oQuoteStandardCite();
-		@include oQuoteStandardCiteAuthor();
-	}
-
+	@include oQuote($opts: (cite: true, standard: true));
```
Comment (chee) [inline scss/_blockquotes.scss]: i think for this component, we'd like to use the other placeholder mixins. Sorry, this isn't very well documented and that's my fault.

for this i'd

```
.n-content-blockquote {
	@include oQuoteBase();
	@include oQuoteStandard();

	cite {
		@include oQuoteBaseCite();
		@include oQuoteStandardCite();
		@include oQuoteStandardCiteAuthor();
	}

	margin-bottom: oSpacingByName('s8');
}
```

File: scss/_body.scss [hunk 4]
```diff
@@ -21,8 +21,8 @@
 
 	// DIVIDER
 	> hr {
-		// margin considers p margin bottom of 28px and hr border size of 4px to equal 36px on both sides
-		@include oTypographyMargin($top: 2, $bottom: 8);
+		margin-bottom: oSpacingByName('s8');
+		margin-top: oSpacingByName('s2');
```
Comment (chee) [inline scss/_body.scss]: neat ^_^

File: scss/_mixins.scss [hunk 5]
```diff
@@ -65,58 +65,21 @@
...
-	&:hover,
-	&:focus {
-		background: rgba(oColorsGetColorFor(#{$use-case}-hover, background), 0.1);
-		border-bottom-width: 0;
-		&:after {
-			@include oIconsGetIcon('arrow-right', oColorsGetColorFor(#{$use-case}-hover, border), 10);
-		}
-	}
-}
-
```
Comment (chee) [inline scss/_mixins.scss]: was this never used?

File: scss/_pullquotes.scss [hunk 6]
```diff
@@ -1,26 +1,27 @@
 .n-content-pullquote {
 	.n-content-body > & {
 		@include nLayoutInset;
-		@include oTypographyPadding($top: 0, $bottom: 0);
+		padding-top: 0;
+		padding-bottom: 0;
```
Comment (chee) [inline scss/_pullquotes.scss]: :heart:

File: test/integration/compile-main.test.js [hunk 7]
```diff
@@ -11,6 +10,7 @@ async function compile({ file, includePaths }) {
 			includePaths
 		}, function(err, result) {
 			if (err) {
+				console.log(err.formatted)
```
Comment (chee) [inline test/integration/compile-main.test.js]:
```suggestion
				console.error(err.formatted)
```

File: test/integration/entry-file.scss [hunk 8]
```diff
@@ -0,0 +1,3 @@
+$system-code: n-content-body;
+
+@import "../../main";
```
Comment (chee) [inline test/integration/entry-file.scss]: nice ^_^

i like this solution

File: scss/_body.scss [hunk 9]
```diff
@@ -32,14 +32,12 @@
 	// LISTS
 	> ul,
 	.n-content-layout__slot > ul {
-		@include oTypographyList();
-		@include oTypographyListUnordered();
+		@include oTypographyList($type: 'unordered');;
```
Comment (i-like-robots) [inline scss/_body.scss]: `;;` 🔍

File: scss/_mixins.scss [hunk 10]
```diff
@@ -7,26 +7,26 @@
...
 			}
 		}
 		&:hover {
-			color: getColor('black-80');
+			color: oColorsByName('black-80');
 			text-decoration: none;
 		}
 		&[aria-expanded="true"] i {
-			@include oIconsGetIcon('arrow-up', getColor('teal-40'), 12);
+			@include oIconsContent('arrow-up', oColorsByName('teal-40'), 12);
```
Comment (i-like-robots) [inline scss/_mixins.scss]: I really want to swap this with `rotate(180deg)`

File: scss/_related-box.scss [hunk 11]
```diff
@@ -34,11 +34,11 @@
 
 .n-content-related-box__headline,
 .n-content-info-box__headline {
-	@include oTypographySansBold($scale: 4, $progressive: true);
+	@include oTypographySans($weight: 'bold', $scale: 4);
 	font-size: 26px;
```
Comment (i-like-robots) [inline scss/_related-box.scss]: Setting the size/line-height twice here

File: scss/_mixins.scss (hunk 5 above)
Comment (edds) [reply scss/_mixins.scss]: Nope https://github.com/search?q=org%3AFinancial-Times+nListsItem&type=Code

File: scss/_related-box.scss (hunk 11 above)
Comment (chee) [reply scss/_related-box.scss]: could we run it by design to see if they'd agree to an on-piste sizing?

File: scss/_mixins.scss (hunk 5 above)
Comment (chee) [reply scss/_mixins.scss]: nice, love too delete the unused mixin

File: scss/_mixins.scss (hunk 10 above)
Comment (notlee) [reply scss/_mixins.scss]: Screenshot 2019-11-26 at 15 11 03

File: scss/_related-box.scss (hunk 11 above)
Comment (edds) [reply scss/_related-box.scss]: I've opened an issue to track this, as it's not a regression I'm going to push on without it.

Comment (chee) [reply scss/_related-box.scss]: that seems like the right choice

**Pull request merged**